src/resources/scripts/ptp.py

```diff
@@ -0,0 +1,835 @@
+def get_point_contour(contours: list, point: np.array) -> Union[list, None]:
+    """
+    Get the contour that contains the desired point
+
+    Args:
+        contours: List of possible contours
+        point: Point used to filter a valid contour
+
+    Returns:
+       A list containing the coordinates for a valid contour containing the point. Return None if not found.
+    """
+    for contour in contours:
+        # check if the point is inside the contour
+        if cv2.pointPolygonTest(contour, point, False) >= 0:
+            return contour
+    # if no contour contains the point return None
+    return None
+
+
+# convert a mask to a contour
+# scores is a necessary input when the multimask input is True.
+def mask_to_contour(
+    masks: np.array,
+    img_height: int,
+    img_width: int,
+    point: np.array,
+    scores: list = [],
+    multimask: bool = True,
+    typical_area: int = 9999999999,
+) -> list:
+    """
+    Converts the predicted mask to a contour
+    Args:
+        masks: Resulting mask from the SAM prediction
+        img_height: Image height
+        img_width: Image width
+        point: The input point annotation
+        scores: Prediction scores from SAM prediction
+        multimask: Whether the prediction is multimask
+        typical_area: Expected prediction area
+
+    Returns:
+        List of coordinates of predictions if valid. List of None otherwise
+    """
+    if multimask is True:
+        assert (
+            len(scores) != 0
+        ), "Scores must be a list from SAM, but the input is an empty list."
+        # init results array
+        results = []
+        # go through all 3 masks of multimask
+        for i in range(3):
+            # and gather the contours and area using recursion
+            results.append(
+                mask_to_contour(masks[i], img_height, img_width, point, scores, False)
+            )
+        # if the typical area is not the default value...
+        if typical_area is not None:
+            # get the absolute differences between the typical area and the area of each mask
+            absdiff = [np.abs(x[1] - typical_area) for x in results if x[1] is not None]
+            if len(absdiff) == 0:
+                return [None, None]
+            # take the one mask that is closest to the typical area
+            idx = np.argmin(absdiff)
+            # and return it
+            return results[idx]
+        else:
+            # otherwise get the one with the best score
+            #TODO: sometimes here it breaks
+            idx = np.argsort(scores)[::-1]
+            mask = np.reshape(masks[idx[0]], (img_height, img_width))
+    else:
+        mask = masks
+    # check if mask contains anything. If not return because no contours can be found anyway
+    if not mask.any():
+        return [None, None]
+    mask = mask * 255
+    mask = mask.astype(np.uint8)
+    contours, hierarchy = cv2.findContours(
+        mask, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_NONE
```

Comment:
We don't want to store coordinates for each pixel :wink: 

```suggestion
        mask, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE
```